services/identity/api.go

```diff
@@ -0,0 +1,226 @@
+		return err
+	}
+	_, err = out.Write(data)
+	return err
+}
+
+// GetProposed returns the Propose-field or a copy of the config if
+// the Propose-field is nil
+func (i *Identity) GetProposed() *Config {
+	if i.Proposed != nil {
+		return i.Proposed
+	}
+	return i.Config.Copy()
+}
+
+// AttachToIdentity proposes to attach it to an existing Identity
+func (i *Identity) AttachToIdentity(ID ID) error {
+	i.ID = ID
+	err := i.ConfigUpdate()
+	if err != nil {
+		return err
+	}
+	if _, exists := i.Config.Device[i.ManagerStr]; exists {
+		return errors.New("Adding with an existing account-name")
+	}
+	confPropose := i.Config.Copy()
+	confPropose.Device[i.ManagerStr] = &Device{i.Public}
+	err = i.ProposeSend(confPropose)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// CreateIdentity asks the identityService to create a new Identity
+func (i *Identity) CreateIdentity() error {
+	msg, err := i.Send(i.Cothority.GetRandom(), &AddIdentity{i.Config, i.Cothority})
+	if err != nil {
+		return err
+	}
+	air := msg.Msg.(AddIdentityReply)
+	i.root = air.Root
+	i.data = air.Data
+	i.ID = ID(i.data.Hash)
+
+	return nil
+}
+
+// ProposeSend sends the new proposition of this identity
+// ProposeVote
+func (i *Identity) ProposeSend(il *Config) error {
+	_, err := i.Send(i.Cothority.GetRandom(), &ProposeSend{i.ID, il})
+	i.Proposed = il
+	return err
+}
+
+// ProposeFetch verifies if there is a new configuration awaiting that
+// needs approval from clients
+func (i *Identity) ProposeFetch() error {
+	msg, err := i.Send(i.Cothority.GetRandom(), &ProposeFetch{
+		ID:          i.ID,
+		AccountList: nil,
+	})
+	if err != nil {
+		return err
+	}
+	cnc := msg.Msg.(ProposeFetch)
+	i.Proposed = cnc.AccountList
+	return nil
+}
+
+// ProposeVote calls the 'accept'-vote on the current propose-configuration
+func (i *Identity) ProposeVote(accept bool) error {
+	if i.Proposed == nil {
+		return errors.New("No proposed config")
+	}
+	log.Lvlf3("Voting %t on %s", accept, i.Proposed.Device)
+	if !accept {
+		return nil
+	}
+	hash, err := i.Proposed.Hash()
+	if err != nil {
+		return err
+	}
+	sig, err := crypto.SignSchnorr(network.Suite, i.Private, hash)
+	if err != nil {
+		return err
+	}
+	msg, err := i.Send(i.Cothority.GetRandom(), &ProposeVote{
+		ID:        i.ID,
+		Signer:    i.ManagerStr,
+		Signature: &sig,
+	})
+	err = sda.ErrMsg(msg, err)
+	if err != nil {
+		return err
+	}
+	sb, ok := msg.Msg.(skipchain.SkipBlock)
+	if ok {
+		log.Lvl2("Threshold reached and signed")
+		i.data = &sb
+		i.Config = i.Proposed
+		i.Proposed = nil
+	} else {
+		log.Lvl2("Threshold not reached")
+	}
+	return nil
+}
+
+// ConfigUpdate asks if there is any new config available that has already
+// been approved by others and updates the local configuration
+func (i *Identity) ConfigUpdate() error {
+	if i.Cothority == nil || len(i.Cothority.List) == 0 {
+		return errors.New("Didn't find any list in the cothority")
+	}
+	msg, err := i.Send(i.Cothority.GetRandom(), &ConfigUpdate{ID: i.ID})
```

Comment:
Well, we need more than 2/3 to avoid a split in our BFT-group, so we would have to add Philipp, and then only being able to vote if 3/4 come together...

Instead of `GetRandom`, what about `RandomSI`, `RandomServerIdentity`, `ChoseRandomly`?